x-pack/plugins/security_solution/cypress/upgrade_integration/threshold_rule.spec.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HOST_NAME, REASON, RISK_SCORE, RULE_NAME, SEVERITY } from '../screens/alerts';
+import { SERVER_SIDE_EVENT_COUNT } from '../screens/alerts_detection_rules';
+import {
+  ADDITIONAL_LOOK_BACK_DETAILS,
+  ABOUT_DETAILS,
+  ABOUT_RULE_DESCRIPTION,
+  CUSTOM_QUERY_DETAILS,
+  DEFINITION_DETAILS,
+  getDetails,
+  INDEX_PATTERNS_DETAILS,
+  RISK_SCORE_DETAILS,
+  RULE_NAME_HEADER,
+  RULE_TYPE_DETAILS,
+  RUNS_EVERY_DETAILS,
+  SCHEDULE_DETAILS,
+  SEVERITY_DETAILS,
+  THRESHOLD_DETAILS,
+  TIMELINE_TEMPLATE_DETAILS,
+} from '../screens/rule_details';
+
+import { expandFirstAlert } from '../tasks/alerts';
+import { waitForPageToBeLoaded } from '../tasks/common';
+import { waitForRulesTableToBeLoaded, goToRuleDetails } from '../tasks/alerts_detection_rules';
+import { loginAndWaitForPage } from '../tasks/login';
+
+import { DETECTIONS_RULE_MANAGEMENT_URL } from '../urls/navigation';
+import {
+  OVERVIEW_HOST_NAME,
+  OVERVIEW_RISK_SCORE,
+  OVERVIEW_RULE,
+  OVERVIEW_SEVERITY,
+  OVERVIEW_STATUS,
+  OVERVIEW_THRESHOLD_COUNT,
+  OVERVIEW_THRESHOLD_VALUE,
+  SUMMARY_VIEW,
+} from '../screens/alerts_details';
+
+const EXPECTED_NUMBER_OF_ALERTS = '1';
+
+const alert = {
+  rule: 'Threshold rule',
+  severity: 'medium',
+  riskScore: '17',
+  reason: 'event created medium alert Threshold rule.',
+  hostName: 'security-solution.local',
+  thresholdCount: '2',
+};
+
+const rule = {
+  customQuery: '*:*',
+  name: 'Threshold rule',
+  description: 'Threshold rule for testing upgrade',
+  index: ['auditbeat-threshold*'],
+  severity: 'Medium',
+  riskScore: '17',
+  timelineTemplate: 'none',
+  runsEvery: '60s',
+  lookBack: '2999999m',
+  timeline: 'None',
+  thresholdField: 'host.name',
+  threholdValue: '1',
+};
+
+describe('After an upgrade, the threshold rule', () => {
+  before(() => {
+    loginAndWaitForPage(DETECTIONS_RULE_MANAGEMENT_URL);
+    waitForRulesTableToBeLoaded();
+    goToRuleDetails();
+    waitForPageToBeLoaded();
+  });
+
+  it('Has the expected alerts number', () => {
+    cy.get(SERVER_SIDE_EVENT_COUNT).contains(EXPECTED_NUMBER_OF_ALERTS);
```

Comment:
This should be an assertion, correct?

```suggestion
    cy.get(SERVER_SIDE_EVENT_COUNT).should('contain', EXPECTED_NUMBER_OF_ALERTS);
```